### PR TITLE
Added Command (Completed the TODO)

### PR DESCRIPTION
**Pull request by 0x41c**
Added a command that allows you to toggle `--on` and `--off` using `.silent-type`
It works right away too! I will be willing to make this into settings but you can already turn commands off and on using powercords `Plugins > Silent Type > Toggle`

Edit: would also like to clarify that the "2 people" were me. I just messed up a new github install

### CODE REVIEW

**Review by kablekompany — COMMENTED** [review]
I believe for this method you'll need to pull in component settings as a constant and let it set the switch (which you're doing already, but it's not actually referencing or injecting a settings conf
`powercord.api.settings.registerSettings` maybe? 
unsure but i can't get it to toggle. Turns off and stays off (showing typing notifs)

File: index.js [hunk 1]
```diff
@@ -1,11 +1,55 @@
...
+                return "I am already turned on\nIf you would like to turn me off use the `--off` flag"
+            }
+            this.settings.set('on')
+            this.callRefresh()
+            return "Turned On"
+        } else if (args.includes('--off')) {
+            if (!this.settings.get('on')) {
+                return "I am already turned off\nIf you would like to turn me on use the `--on` flag"
+            }
+            this.settings.set('on')
```
Comment (kablekompany) [inline index.js]: `this.settings.set(false)`

For feasibility sake I'd like to change I/O to a Boolean. Easier to manipulate if a settings panel is added later too

File: index.js [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+    }
+     handler(args) {
+        args = args.map(a => a.toLowerCase())
+        if (args.includes('--off') && args.includes('--on') || args.includes('--on') && args.includes('status')) {
+            return "Oops! Instead of receiving one flag I got both!\nSend only one flag next time"
+        } else if (args.includes('--on')) {
+            if (this.settings.get('on', true)) {
+                return "I am already turned on\nIf you would like to turn me off use the `--off` flag"
+            }
+            this.settings.set('on')
```
Comment (kablekompany) [inline index.js:23]: `this.settings.set(true)`

see next line comment re: boolean indication I/O

File: index.js [hunk 3]
```diff
@@ -1,11 +1,55 @@
...
+        } else {
+            return `Incorrect Flags or no flags given\nPlease run \`${powercord.api.commands.prefix}help silent-type\` for command usage`
+        }
+    }
+
+    callRefresh() {
+        try {
+            const { typing } = require('powercord/webpack');
+            typing.startTyping = (startTyping => (channel) => setImmediate(() => {
+                const on = this.settings.get('on')
```
Comment (kablekompany) [inline index.js:36]:
```suggestion
                const on = this.settings.get()
```

**Review by kablekompany — CHANGES_REQUESTED** [review]
I think since you check on in configurations for true, that is our stationary setter. And true is on false is off. I've noted the recommendation. Let me know if you agree.

**Comment by 0x41c** [comment]
```javascript
this.settings.set()
```
This is an api thing used with multiple settings. I could reassign it but it has no actual effect to the actual performance.
My latest commit should be all thats needed for this plugin unless you just nEeD settings but I doubt that lol

**Comment by kablekompany** [comment]
Sweet. I'll test and merge if all works out. Will be at injected client in a few hours.

**Comment by kablekompany** [comment]
LGTM 
tested and works 
Thanks for the contribution

**Pull request merged**